gtsfm/data_association/point3d_initializer.py

```diff
@@ -58,7 +58,87 @@ class Point3dInitializer(NamedTuple):
     reproj_error_thresh: float
     num_ransac_hypotheses: Optional[int] = None
 
-    def triangulate(self, track_2d: SfmTrack2d) -> Optional[SfmTrack]:
+
+    def execute_ransac_variant(self, track_2d: SfmTrack2d) -> np.ndarray:
+        """ Execute RANSAC algorithm to find best subset 2d measurements for a 3d point
+
+        Args:
+            track: feature track with N 2d measurements in separate images
+
+        Returns:
+            best_inliers: boolean array of length N. Indices of measurements
+               are set to true if they correspond to the best RANSAC hypothesis
+        """
+        # Generate all possible matches
+        measurement_pairs = self.generate_measurement_pairs(track_2d)
+
+        # limit the number of samples to the number of available pairs
+        num_hypotheses = min(self.num_ransac_hypotheses, len(measurement_pairs))
+
+        # Sampling
+        samples = self.sample_ransac_hypotheses(track_2d, measurement_pairs, num_hypotheses)
+
+        # Initialize the best output containers
+        best_num_votes = 0
+        best_error = MAX_TRACK_REPROJ_ERROR
+        best_inliers = np.zeros(len(track_2d.measurements), dtype=bool)
+
+        for sample_idxs in samples:
+            k1, k2 = measurement_pairs[sample_idxs]
+
+            i1, uv1 = track_2d.measurements[k1]
+            i2, uv2 = track_2d.measurements[k2]
+
+            camera_estimates = CameraSetCal3Bundler()
+            # check for unestimated cameras
+            if self.track_camera_dict.get(i1) != None and self.track_camera_dict.get(i2) != None:
+                camera_estimates.append(self.track_camera_dict.get(i1))
+                camera_estimates.append(self.track_camera_dict.get(i2))
+
+                img_measurements = Point2Vector()
+                img_measurements.append(uv1)
+                img_measurements.append(uv2)
+
+                # triangulate point for track
+                try:
+                    triangulated_pt = gtsam.triangulatePoint3(
+                        camera_estimates,
+                        img_measurements,
+                        rank_tol=SVD_DLT_RANK_TOL,
+                        optimize=True,
+                    )
+                except RuntimeError:
+                    # TODO: handle cheirality exception properly?
+                    logger.info(
+                        "Cheirality exception from GTSAM's triangulatePoint3() likely due to outlier, skipping track"
+                    )
+                    continue
+
+                errors = self.compute_track_reprojection_errors(track_2d.measurements, triangulated_pt)
```

Comment:
I think it would be more modular and cleaner if you made the above changes, and moved this code to a different function `compute_reprojection_error_for_hypothesis(track_2d.measurements, k1, k2)`

Reply:
@akshay-krishnan can you point me to which line you were referring to?

Reply:
I was referring to all the lines I selected. Basically, the code for computing the errors for each sample_idx could be moved to a different function as this function is getting a little too big. But it's fine if you'd like to keep it this way.